### PR TITLE
Refactoring: Extracts a few functions

### PR DESCRIPTION
**Pull request by Glinkis**
I have extracted a few more functions, and added jsdoc-style documentation comments to them.

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -100,12 +103,7 @@ function initializeBoard() {
 function activateBoard() {
 	logger.log("Entered into activateBoard");
 
-	for (var row = 1; row <= 3; row++) {
-		for (var col = 1; col <= 3; col++) {
-			logger.log("Adding listener for row[" + row + "] col[" + col + "]");	
-			document.getElementById("cell-" + row + col).addEventListener("click", playerMove, false);
-		}
-	}
+	addCellListeners();
```
Comment (satyapavan) [inline js/script.js:130]: Hi @Glinkis , I think it's better to rename the function to "activateGrid" or "activateCells" as it goes well with the parent function naming

File: js/game_state.js [hunk 2]
```diff
@@ -308,4 +290,43 @@ var cGameState = function (level) {
...
+
+/**
+ * Copies all the cells from one board to another.
+ * @param {string[][]} from - Board to copy cells from.
+ * @param {string[][]} to  - Board to copy cells to.
+ */
+function cloneCells(from, to) {
+	for (var row = 0; row < 3; row++) {
+		for (var col = 0; col < 3; col++) {
+			cloneTo[row][col] = cloneFrom[row][col];
```
Comment (satyapavan) [inline js/game_state.js]: The function parameters are not matching with the implementations.
`from `vs. `cloneFrom `and `to `vs `cloneTo`.  Can you please invoke them in your local and see for further issues?

**Review by satyapavan — CHANGES_REQUESTED** [review]

File: js/game_state.js (hunk 2 above)
Comment (Glinkis) [reply js/game_state.js]: Sorry about this. I'm used to intelligent renaming which is doesn't really work with this. I'll fix it ASAP.

File: js/script.js (hunk 1 above)
Comment (Glinkis) [reply js/script.js:130]: Agreed.

File: js/game_state.js [hunk 3]
```diff
@@ -308,4 +290,43 @@ var cGameState = function (level) {
...
+			to[row][col] = from[row][col];
+		}
+	}
+}
+
+/**
+ * @param {string[][]} board
+ * @param {[number, number][]} finish_moves
+ */
+function checkDiagonalA(board, finish_moves) {
```
Comment (satyapavan) [inline js/game_state.js:312]: Hi @Glinkis , These 3 new functions are outside the GameState scope. As they do not server any porpose outside the scope, would you like to move them inside?

File: js/game_state.js [hunk 4]
```diff
@@ -298,9 +298,9 @@ var cGameState = function (level) {
  * @param {string[][]} to  - Board to copy cells to.
  */
 function cloneCells(from, to) {
```
Comment (satyapavan) [inline js/game_state.js]: in `cloneCells` the first argument is `from `and 2nd is `to `, but while invoking we are passing it the other way which is leading to failure

**Review by satyapavan — CHANGES_REQUESTED** [review]

File: js/game_state.js (hunk 3 above)
Comment (Glinkis) [reply js/game_state.js:312]: Seems I can't edit this any more after deleting my fork... I'll close this PR.

Comment (satyapavan) [reply js/game_state.js:312]: oh! too bad.

**Pull request closed without merging**